x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -244,7 +244,7 @@ public IndexInput openInput(final String name, final IOContext context) throws I
         if (useCache && isExcludedFromCache(name) == false) {
             return new CachedBlobContainerIndexInput(this, fileInfo, context, inputStats);
         } else {
-            return new DirectBlobContainerIndexInput(blobContainer, fileInfo, context, uncachedChunkSize, BufferedIndexInput.BUFFER_SIZE);
+            return new DirectBlobContainerIndexInput(this, fileInfo, context, inputStats, uncachedChunkSize, bufferSize(context));
```

Comment:
I noticed this while writing tests: we don't align the buffer size in the same way than BufferedIndexInput would do (ie, based on the IOContext)

Reply:
Good catch :+1: